### ISSUE TITLE
Router named outlet in child route of an empty path doesn't work

### ISSUE THREAD

**Issue by ernestbofill** [angular/angular#27623]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/router

### Is this a regression?

 No

### Description

 I have a route with empty path, and in it's children I would like to define a named outlet route in the following way.

```ts
{
  path: '',
  children: [
    {
      path: 'mywidget',
      component: MyWidgetComponent,
      outlet: 'widget',
    },
    {
      path: '',
      component: MainPageComponent,
    }
  ]
},
{
  path: 'other',
  component: OtherPageComponent
},
```

This config fails with the following error.

```
Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'mywidget'
Error: Cannot match any routes. URL Segment: 'mywidget'
```

Giving a path name to the parent route solve the problem, but I would like to use an empty path there because it's a default route.

```ts
{
  path: 'main',
  children: [
    {
      path: 'mywidget',
      component: MyWidgetComponent,
      outlet: 'widget',
    },
    {
      path: '',
      component: MainPageComponent,
    }
  ]
},
{
  path: 'other',
  component: OtherPageComponent
},
```

## 🔬 Minimal Reproduction

 
Not working:
[https://stackblitz.com/edit/angular-named-outlet-empty-path](https://stackblitz.com/edit/angular-named-outlet-empty-path)

And this is the fix that I'm trying to avoid:
[https://stackblitz.com/edit/angular-routing-child-named-outlet](https://stackblitz.com/edit/angular-routing-child-named-outlet)

## 🔥 Exception or Error
<pre><code>Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'mywidget'
Error: Cannot match any routes. URL Segment: 'mywidget'
at ApplyRedirects.noMatchError (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/@angular/router@7.1.3/bundles/router.umd.js:2557:20)
at CatchSubscriber.eval [as selector] (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/@angular/router@7.1.3/bundles/router.umd.js:2538:33)
at CatchSubscriber.error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/operators/catchError.js:48:31)
at MapSubscriber.Subscriber._error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:93:26)
at MapSubscriber.Subscriber.error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:73:18)
at MapSubscriber.Subscriber._error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:93:26)
at MapSubscriber.Subscriber.error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:73:18)
at MapSubscriber.Subscriber._error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:93:26)
at MapSubscriber.Subscriber.error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/Subscriber.js:73:18)
at TapSubscriber._error (https://angular-named-outlet-empty-path.stackblitz.io/turbo_modules/rxjs@6.3.3/internal/operators/tap.js:75:26)</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>@angular-devkit/architect         0.7.5
@angular-devkit/build-angular     0.7.5
@angular-devkit/build-optimizer   0.7.5
@angular-devkit/build-webpack     0.7.5
@angular-devkit/core              0.7.5
@angular-devkit/schematics        0.7.5
@angular/cli                      6.1.5
@angular/platform-server          7.0.2
@ngtools/webpack                  6.1.5
@schematics/angular               0.7.5
@schematics/update                0.7.5
rxjs                              6.3.3
typescript                        2.7.2
webpack                           4.9.2</code></pre>

**Comment by TGC79** [comment]
Duplicate of #10726 which should be reopened since the fix for it was reverted.

Issue was first reported on Angular 2.0.0-rc.5.

**Comment by DaSchTour** [comment]
This is still an issue. Don't close this like https://github.com/angular/angular/issues/10726

**Comment by nedim1511** [comment]
Still an issue in 2020.

**Comment by DaSchTour** [comment]
Just as as reminder, this closed duplicate #10726 had 88 votes!

**Comment by atscott** [comment]
Updating the example to v12 shows that this is working. If there’s a scenario that isn’t, please open a new issue with a new reproduction.
